### DISCUSSION TITLE
Adding a coap resource with uri params

### DISCUSSION THREAD

**Discussion by Exormeter** [openthread/openthread#10030] [Q&A]
Hello, I have a coap resource which looks like this:

```
{
      .mContext = NULL,
      .mHandler = handleNodeIdResolve,
      .mNext = NULL,
      .mUriPath = "i/r"
 }
```

I can call the resource like this with no problem _[serverip] i/r_.
Now I would like to add a uri parameter to my resource to call it like this _[serverip] i/r?test=test_,
but when I add the parameter to the uri, my handler is no longer called. What do I need to do so I can 
add a uri parameter?

**Comment by jwhui** [comment]
There is an open PR to add support for CoAP URI queries via the CLI:
- https://github.com/openthread/openthread/pull/10003

**Reply by Exormeter** [reply]
I have seen the open PR for CoAP URI queries, but my question is about the server side. I would like to know how I have to add a coap resource in the server that supports URI params.

**Reply by jwhui** [reply]
This would require updating the existing OpenThread implementation. Contributions are welcome!
